pulseaudio-dlna: unstable-2017-11-01 -> unstable-2021-11-09

flokli opened the pull request:
This moves pulseaudio-dlna to a more recent fork, which works with
Python 3.

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

flokli committed:
pulseaudio-dlna: unstable-2017-11-01 -> unstable-2021-11-09

This moves pulseaudio-dlna to a more recent fork, which works with
Python 3.

thiagokokada commented:
Please fix the commit and PR title:

```
 pulseaudio-dlna: unstable-2017-11-01 -> unstable-2021-11-019
```

flokli commented:
@thiagokokada thanks for spotting :laughing: It seems I copied the same line into the commit message ;-)

thiagokokada commented:
Got the following error while running:

```
Traceback (most recent call last):
  File "/nix/store/fkzla307l4mlcvfyshsrccwl7szihx2z-python3-3.9.6/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/nix/store/fkzla307l4mlcvfyshsrccwl7szihx2z-python3-3.9.6/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/nix/store/jwz6d7273pif6v1g7r0dskcls0x6jw56-pulseaudio-dlna-unstable-2021-11-09/lib/python3.9/site-packages/pulseaudio_dlna/pulseaudio.py", line 617, in run
    self._connect(signals)
  File "/nix/store/jwz6d7273pif6v1g7r0dskcls0x6jw56-pulseaudio-dlna-unstable-2021-11-09/lib/python3.9/site-packages/pulseaudio_dlna/pulseaudio.py", line 55, in _connect
    self.bus = self._get_bus()
  File "/nix/store/jwz6d7273pif6v1g7r0dskcls0x6jw56-pulseaudio-dlna-unstable-2021-11-09/lib/python3.9/site-packages/pulseaudio_dlna/pulseaudio.py", line 138, in _get_bus
    modules = self.get_modules()
  File "/nix/store/jwz6d7273pif6v1g7r0dskcls0x6jw56-pulseaudio-dlna-unstable-2021-11-09/lib/python3.9/site-packages/pulseaudio_dlna/pulseaudio.py", line 233, in get_modules
    process = subprocess.Popen(
  File "/nix/store/fkzla307l4mlcvfyshsrccwl7szihx2z-python3-3.9.6/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/nix/store/fkzla307l4mlcvfyshsrccwl7szihx2z-python3-3.9.6/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'pactl'
```

I think you need to add `pulseaudio` to `buildInputs` :thinking: ?

thiagokokada commented on pkgs/applications/audio/pulseaudio-dlna/0001-setup.py-remove-dbus-python-from-list.patch:
```diff
@@ -0,0 +1,25 @@
+     install_requires=[
+     'docopt',
+     'chardet',
+-    'dbus-python',
```
This seems very strange, maybe someone with more knowledge on how Python works in nixpkgs can help here.

CC @SuperSandro2000 .

flokli replied:
I found https://github.com/PeterZainzinger/pulseaudio-dlna/commit/a8d76f18ded1f57e275cc9cb27a6843268600539 (a fork of this), so we might not be the only ones with this problem.

It really looks like setuptools doesn't properly discover it.

SuperSandro2000 replied:
Not sure, this should just work.

flokli replied:
But you did remove the patch for a quick test, and can confirm it doesn't work without? ;-)

flokli replied:
@SuperSandro2000 it seems https://github.com/NixOS/nixpkgs/pull/119469 (which you reviewed as well) had the same problem. I propose we keep this patch for now.

flokli commented:
> Got the following error while running:
> 
> ```
> Traceback (most recent call last):
>   File "/nix/store/fkzla307l4mlcvfyshsrccwl7szihx2z-python3-3.9.6/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
>     self.run()
>   File "/nix/store/fkzla307l4mlcvfyshsrccwl7szihx2z-python3-3.9.6/lib/python3.9/multiprocessing/process.py", line 108, in run
>     self._target(*self._args, **self._kwargs)
>   File "/nix/store/jwz6d7273pif6v1g7r0dskcls0x6jw56-pulseaudio-dlna-unstable-2021-11-09/lib/python3.9/site-packages/pulseaudio_dlna/pulseaudio.py", line 617, in run
>     self._connect(signals)
>   File "/nix/store/jwz6d7273pif6v1g7r0dskcls0x6jw56-pulseaudio-dlna-unstable-2021-11-09/lib/python3.9/site-packages/pulseaudio_dlna/pulseaudio.py", line 55, in _connect
>     self.bus = self._get_bus()
>   File "/nix/store/jwz6d7273pif6v1g7r0dskcls0x6jw56-pulseaudio-dlna-unstable-2021-11-09/lib/python3.9/site-packages/pulseaudio_dlna/pulseaudio.py", line 138, in _get_bus
>     modules = self.get_modules()
>   File "/nix/store/jwz6d7273pif6v1g7r0dskcls0x6jw56-pulseaudio-dlna-unstable-2021-11-09/lib/python3.9/site-packages/pulseaudio_dlna/pulseaudio.py", line 233, in get_modules
>     process = subprocess.Popen(
>   File "/nix/store/fkzla307l4mlcvfyshsrccwl7szihx2z-python3-3.9.6/lib/python3.9/subprocess.py", line 951, in __init__
>     self._execute_child(args, executable, preexec_fn, close_fds,
>   File "/nix/store/fkzla307l4mlcvfyshsrccwl7szihx2z-python3-3.9.6/lib/python3.9/subprocess.py", line 1821, in _execute_child
>     raise child_exception_type(errno_num, err_msg, err_filename)
> FileNotFoundError: [Errno 2] No such file or directory: 'pactl'
> ```
> 
> I think you need to add `pulseaudio` to `propagatedBuildInputs` thinking ?

`pulseaudio-dlna` picks `pactl` from `$PATH`. We could prefix `$PATH` to add it, but this hasn't changed in this PR - maybe there's valid reasons we don't hardcode this (as people might use different flavours of `pulseaudio`, and `pactl` should use the right one).

thiagokokada commented:
> `pulseaudio-dlna` picks `pactl` from `$PATH`. We could prefix `$PATH` to add it, but this hasn't changed in this PR - maybe there's valid reasons we don't hardcode this (as people might use different flavours of `pulseaudio`, and `pactl` should use the right one).

Yeah, but I think packages should be as hermetic as possible. Specially if this is something that seems to be not optional.

In my case, I am using PipeWare. So while I have full PulseAudio compatibility, I don't have `pactl` in my `PATH` (I could include it though).

But anyway, this is just my vision. Some other reviewer may have theirs. I am fine either way.

SuperSandro2000 commented on pkgs/applications/audio/pulseaudio-dlna/default.nix:
```diff
@@ -1,10 +1,18 @@
+, vorbis-tools ? null
 }:
 
 assert mp3Support -> lame != null;
```
Those can be removed.

SuperSandro2000 commented on pkgs/applications/audio/pulseaudio-dlna/default.nix:
```diff
@@ -1,10 +1,18 @@
+{ fetchFromGitHub
+, lib
+, python3Packages
+, mp3Support ? true
+, lame ? null
+, opusSupport ? true
+, opusTools ? null
+, faacSupport ? false
+, faac ? null
+, flacSupport ? true
+, flac ? null
+, soxSupport ? true
+, sox ? null
+, vorbisSupport ? true
+, vorbis-tools ? null
```
```suggestion
, mp3Support ? true
, lame
, opusSupport ? true
, opusTools
, faacSupport ? false
, faac
, flacSupport ? true
, flac
, soxSupport ? true
, sox
, vorbisSupport ? true
, vorbis-tools
```

flokli committed:
pulseaudio-dlna: minor cleanups

flokli committed:
pulseaudio-dlna: ensure pactl is available

pulseaudio-dlna shells out to pactl to configure sinks and sources.
As pactl might not be in $PATH, add --suffix it (so pactl configured by the
user get priority)

flokli commented:
@thiagokokada I pushed another version, which prefers `pactl` from `$PATH`, then falls back the one provided by `pulseaudio`.

> In my case, I am using PipeWare. So while I have full PulseAudio compatibility, I don't have pactl in my PATH (I could include it though).

Note you can't test this with PipeWire, as it's very specific to `pulseaudio`, and requires more than just a pulseaudio-like interface to play sound to.

thiagokokada commented:
Result of `nixpkgs-review pr 146915` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pulseaudio-dlna</li>
  </ul>
</details>

thiagokokada commented:
Since there doesn't seem to be a better solution, I am going to merge.

Just waiting the eval checks.
